app/models/criteria/abstract_criteria.rb

```diff
@@ -12,13 +14,18 @@ def add_condition(condition)
     def satisfy(relation)
       finalize_conditions
       prepared_relation = prepare_relation(relation)
-      satisfy_relation(prepared_relation).uniq.order(:id)
+      satisfied_relation = satisfy_relation(prepared_relation).uniq
+      @sorter.call(satisfied_relation)
     end
 
     def set_preparation(&block)
       @preparation = block
     end
 
+    def set_sorter(&block)
```

Comment:
__[rubocop]__
* Style/AccessorMethodName: Do not prefix writer method names with `set_`.